### PR TITLE
refactor: unnecessary functions and copying

### PR DESCRIPTION
**Pull request by RaitaroH**
I don't see the point of those `cp` commands. I suggest removing them. Nobody is gonna see the screenshot anyway.

### CODE REVIEW

File: bin/catalina_dynamic [hunk 1]
```diff
@@ -8,24 +8,11 @@
...
-    _img_path="$_base_dir/images/3840x3840.jpg"
-    cp "$_base_dir/images/hourly/$1.jpg" $_img_path
-    cp "$_img_path" "$_base_dir/screenshot.jpg"
-    ~/bin/ksetwallpaper $_img_path
-}
-
 # Keep looping.
 while true; do
-    _hour=$(get_hour)
+    _hour=$(date +%H)
```
Comment (RaitaroH) [inline bin/catalina_dynamic]: this is the same as running the function get_hour

File: bin/catalina_dynamic [hunk 2]
```diff
@@ -8,24 +8,11 @@
...
     if [[ "$_old_hour" != "$_hour" ]]; then
-        set_wallpaper $_hour
+        ~/bin/ksetwallpaper "$_base_dir/images/hourly/$_hour.jpg"
```
Comment (RaitaroH) [inline bin/catalina_dynamic]: idk why you did that way but, the way I have it here sets the wall and modifies no extra files.

**Comment by caglarturali** [comment]
They'll see the screenshot if they open up configure desktop dialog. Basically, those `cp` commands synchronize the current wallpaper to the one user sees on configure desktop dialog. I'd like to keep it that way.

File: bin/catalina_dynamic (hunk 2 above)
Comment (caglarturali) [reply bin/catalina_dynamic]: Explained below. :)

**Comment by RaitaroH** [comment]
> They'll see the screenshot if they open up configure desktop dialog. Basically, those `cp` commands synchronize the current wallpaper to the one user sees on configure desktop dialog. I'd like to keep it that way.

```
ln -fs "$_base_dir/images/hourly/$_hour.jpg" "$_base_dir/screenshot.jpg"
```
Isn't this doing the same then? Link the hourly image to the screenshot one. No need for copy, uses less space on drive, less wear and tear.

File: bin/catalina_dynamic (hunk 2 above)
Comment (RaitaroH) [reply bin/catalina_dynamic]: Countered, further down :P

**Comment by caglarturali** [comment]
> > They'll see the screenshot if they open up configure desktop dialog. Basically, those `cp` commands synchronize the current wallpaper to the one user sees on configure desktop dialog. I'd like to keep it that way.
> 
> ```
> ln -fs "$_base_dir/images/hourly/$_hour.jpg" "$_base_dir/screenshot.jpg"
> ```
> 
> Isn't this doing the same then? Link the hourly image to the screenshot one. No need for copy, uses less space on drive, less wear and tear.

Yeah, this is better. I'll close this one but I'll be merging if you open a new PR.
By the way, I'm not a bash-scripting ninja. So, doing redundant things without any obvious reason is somewhat normal in this case. :)

**Comment by RaitaroH** [comment]
No need for new PR, made a new commit here.

**Comment by caglarturali** [comment]
> No need for new PR, made a new commit here.

Looks like I'm new to GitHub, too. :blush:
Thank you for everything.

**Comment by caglarturali** [comment]
By the way, it turned out that symlink is not doing the same thing.
Screenshot_20190707_001253

**Comment by caglarturali** [comment]
> @RaitaroH By the way, it turned out that symlink is not doing the same thing.
> Screenshot_20190707_001253

I've solved it with [this commit](https://github.com/caglarturali/catalina-dynamic-wallpaper-kde/commit/4ed7307e6d33c5e6dd0897f3b72f58861512d112).

**Pull request merged**